src/main/java/com/mentoria/featuretoggle/domain/model/dto/UserPatchDTO.java

```diff
@@ -0,0 +1,21 @@
+package com.mentoria.featuretoggle.domain.model.dto;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.AssertTrue;
+
+
+@Data
+@NoArgsConstructor
+public class UserPatchDTO {
+    private String name;
+    private String password;
+
+    @AssertTrue(message = "name or password is required")
+    private boolean isNameOrPasswordExists() {
+        return ((this.name != null) && !this.name.isEmpty())
```

Comment:
interessante o jeito que tu fez a validação. Mas esse lógica ae ta muito complexa, pra melhorar, tuo podia usar da forma abaixo. Pra validar String Int e essas coisas sempre tem uma libzinha que facilita.

` !Strings.isBlank(this.name) || !Strings.isBlank(this.password);`

Reply:
Acho que dava pra melhorar um pouquinho o nome do método tbm...

isNameOrPasswordNotBlank ou isNameOrPasswordSetted ou algo assim. Exists achei meio estranho, que tu diz?

Reply:
Provavelmente nomeei esse método olhando o jogo do Brasil KKKKKK